lib/shiika_parser/src/lexer.rs

```diff
@@ -312,11 +312,15 @@ impl<'a> Lexer<'a> {
 
     fn read_separator(&mut self, next_cur: &mut Cursor) -> Token {
         while let CharType::Space | CharType::Separator = self.char_type(next_cur.peek(self.src)) {
-            next_cur.proceed(self.src);
+            let c = next_cur.proceed(self.src);
+            if c == '\n' {
+                return Token::Newline;
+            }
         }
-        Token::Separator
+        Token::Semicolon
```

Comment:
The implementation read_comment returns Token::Newline seems does not have any problem.
But I don't have absolute confidence about this behavior.

If you have any concern about this implementation, please notify me.

Reply:
Looks good. Especially, this will allow us to write comments in a method chain.

```
foo
  .bar  # A comment
  .baz 
```